filebeat/inputsource/tcp/server.go

```diff
@@ -85,3 +86,10 @@ func (s *Server) createServer() (net.Listener, error) {
 	}
 	return l, nil
 }
+
+func (s *Server) network() string {
```

Comment:
This is style/consistenty comment.  I'm ok if you leave it as is, just want to bring it up to consider.

A common theme in the beats code base for default values is to make a defaultConfig function, that returns the struct with the default value filled in.  And then when Unpack is run, if the supplied config didn't have the value, you get the default.  And validate can work on the real values you want to check.  An example would be:

https://github.com/elastic/beats/blob/6d25d46033c371c2c8ba0f98e1c7dd803d404e15/filebeat/input/tcp/input.go#L50-L67

what do you think about using that pattern here?

Reply:
Yeah, I saw this pattern, was not sure if that is consistent across the board and if nobody is going to ask why we do set the configuration option when it's was not set. 

The original ticket requirement was: 
```
"When no option is provided, the default value (udp or tcp) is used"
```

So, took it literally in this case: not changing the config, and using the default network type "When no option is provided". I'd probably still leave this ```network()``` function at the point when the server is created in case if the config value gets messed up on a way to the server initialization.